py/htm/examples/mnist.py

```diff
@@ -107,12 +110,11 @@ def main(parameters=default_parameters, argv=None, verbose=True):
     sp = SpatialPooler(
         inputDimensions            = enc.dimensions,
         columnDimensions           = parameters['columnDimensions'],
-        potentialRadius            = 99999999,
+        potentialRadius            = parameters['potentialRadius'],
         potentialPct               = parameters['potentialPct'],
         globalInhibition           = True,
         localAreaDensity           = parameters['localAreaDensity'],
-        numActiveColumnsPerInhArea = -1,
-        stimulusThreshold          = int(round(parameters['stimulusThreshold'])),
+        stimulusThreshold          = int(round(parameters['stimulusThreshold'])), #param is requested to be an integer, but param optimization might find fractional value, so round it
```

Comment:
I think that the parameter optimization program will round the numbers, if it detects that the input number was an integer.

Reply:
this code has been there, I've just added comment why is it done.

Reply:
btw, does the optimizer distinguish between real and integer values? Is optimizing with rounding/cropping to integer would be really wasteful

Reply:
IIRC Yes, The optimization framework rounds all integers back to integers before giving them to the experiment program.  The framework looks at the default parameters to determine the correct data type for each parameter.

Reply:
> The optimization framework rounds all integers back to integers before giving them to the experiment program.  

that is only the compatibility part of the solution. My point is, if I want to change/swarm an integer param, verify that the integer value actually changed. So it does not try 7.3, ... from 7, but jumps directly to 6/8.

Reply:
It depends on the optimizer code used, but in general yes, the optimizer are smart enough to change integers in increments of at least 1.